src/code/z_play.c

```diff
@@ -489,98 +501,109 @@ void Gameplay_Update(GlobalContext* globalCtx) {
                         }
                     }
 
-                    if (CREG(11) == 0) {
-                        func_800BC5E0(globalCtx, globalCtx->fadeTransition);
+                    if (!R_TRANS_DBG_ENABLED) {
+                        Gameplay_SetupTransition(globalCtx, globalCtx->transitionType);
                     } else {
-                        func_800BC5E0(globalCtx, CREG(12));
+                        Gameplay_SetupTransition(globalCtx, R_TRANS_DBG_TYPE);
                     }
 
-                    if (globalCtx->transitionMode >= 4) {
+                    if (globalCtx->transitionMode >= TRANS_MODE_FILL_WHITE_INIT) {
+                        // special modes break out of this switch, instance types continue
```

Comment:
```suggestion
                        // special modes break out of this switch, instance modes continue
```
also I don't think the "special modes" terminology is used anywhere else, so it may be better to refer to those as "non-instance modes"?